analyzers/src/SonarAnalyzer.Common/Rules/CommentsShouldNotBeEmptyBase.cs

```diff
@@ -0,0 +1,126 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class CommentsShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4663";
+
+    protected abstract string GetCommentText(SyntaxTrivia trivia);
+    protected abstract bool IsValidTriviaType(SyntaxTrivia trivia);
+
+    protected abstract bool IsSimpleComment(SyntaxTrivia trivia);
+    protected abstract bool IsEndOfLine(SyntaxTrivia trivia);
+    protected abstract bool IsWhitespace(SyntaxTrivia trivia);
```

Comment:
For consistency: Add these to ISyntaxKindFacade too.

Reply:
Same question as above. :)

Reply:
Same answer ;-)

Reply:
Here, i also can't seem to do the following casts:

```csharp    
protected override bool IsTargetThingy(SyntaxTrivia trivia) =>
        (TSyntaxKind)trivia.RawKind == Language.SyntaxKind.TargetThingy;
```
or even:
```csharp    
protected override bool IsTargetThingy(SyntaxTrivia trivia) =>
        trivia.RawKind == (int)Language.SyntaxKind.TargetThingy;
```

Do you think we should add them to the Facade anyway? Or should I leave them as they are, since I can't leverage the base class?

Reply:
Discussed internally